src/back/TicketManager/src/Core/TicketManager.Core.Services/Endpoints/Tickets/TicketBuyEndpoint.cs

```diff
@@ -27,15 +31,44 @@ public override void Configure()
 
     public override async Task HandleAsync(TicketBuyRequest req, CancellationToken ct)
     {
-        // This code really doesn't matter for now, it can wait for further tasks.
         var @event = await coreDbContext
             .Events
             .Where(e => e.Id == req.EventId)
             .FirstOrDefaultAsync(ct);
         var sector = @event!.Sectors.FirstOrDefault(s => s.Name == req.SectorName);
+
+        var sectorReservationId = await coreDbContext.SectorReservations
+            .Where(sr => sr.EventId == req.EventId && sr.SectorName == req.SectorName).Select(sr => sr.Id)
+            .FirstOrDefaultAsync(ct);
+        
+        bool updateRepository = true;
+        var sectorReservation = await sectorReservationRepository.FindAsync(sectorReservationId, ct);
+
+        if (sectorReservation is null)
+        {
+            sectorReservation = new SectorReservation(req.EventId, req.SectorName);
+            updateRepository = false;
+        }
         
-        // TODO: lock seats
+        var freeSeats = sector!.NumberOfSeats - sectorReservation.SeatReservations.Sum(sr => sr.ReservedSeatNumber);
         
+        if (freeSeats < req.NumberOfSeats)
```

Comment:
Are you sure it will work with what we planned for concurrency? If a new `SeatReservation` is added after `Validator` but before `HandleAsync` won't the `freeSeats` change?